pkg/snapshot/snapshot.go

```diff
@@ -108,12 +124,24 @@ func (s *Snapshotter) snapshotFiles(f io.Writer, files []string) (bool, error) {
 		if err != nil {
 			return false, err
 		}
-		// Only add to the tar if we add it to the layeredmap.
-		addFile, err := s.l.MaybeAdd(file)
+
+		var fileAdded bool
+		lastParentFileIndex := len(files) - n
+		isParentDir := i < lastParentFileIndex
```

Comment:
this is one of the grossest things ive written in a long time (*knock on wood*) @priyawadhwa so let me know if it's not clear and/or you'd just like it to be nicer

i think the next step to making this better would be to break some of this logic out into a separate function or two

Reply:
kk let's do it!